App/Services/IRepository.cs

```diff
@@ -2,12 +2,12 @@
 {
     interface IRepository
     {
-        int AddTask(Task newTask);
-        void DeleteTask(int task_id);
-        void UpdateTaskTitle(int task_id, string newTitle);
-        void UpdateTaskDescription(int task_id, string newDescription);
-        void UpdateTaskDueTime(int task_id, DateTimeOffset newDue);
-        Task GetTask(int task_id);
+        long? AddTask(Task newTask);
```

Comment:
I think you over-copypasted here.
It doesn't make sense to delete the task with `null` task_id]

Change all `long? task_id` arguments to just `long task_id`